table-filter.md

```diff
@@ -0,0 +1,252 @@
+* U+1F926 U+1F3FF U+200D U+2640 U+FE0F (🤦🏿‍♀️) are 5 characters.
+
+### File import
+
+Include an `@` at the beginning of the rule to specify a file name. The table filter parser treats each line of the imported file as additional filter rules.
+
+For example, if a file `config/filter.txt` has content:
+
+```
+employees.*
+*.WorkOrder
+```
+
+the following two invocations are equivalent:
+
+```bash
+./dumpling -f '@config/filter.txt'
+./dumpling -f 'employees.*' -f '*.WorkOrder'
+```
+
+A filter file cannot further import another file.
+
+### Comments and blank lines
+
+Inside a filter file, leading and trailing white-spaces of every line are trimmed. Furthermore, blank lines (empty strings) are ignored.
+
+A leading `#` marks a comment and is ignored. `#` not at start of line may be considered syntax error.
+
+```
+# this line is a comment
+db.table   # but this part is not comment and may cause error
+```
+
+### Exclusion
+
+An `!` at the beginning of the rule means the pattern after it is used to exclude tables from being processed. This effectively turns the filter into a block list.
+
+```
+*.*
+#^ note: must add the *.* to include all tables first
+!*.Password
+!employees.salaries
+```
+
+### Escape character
+
+Precede any special character by a `\` to turn it into an identifier character.
+
+```
+db\.with\.dots.*
+```
+
+For simplicity and future compatibility, the following sequences are prohibited:
+
+* `\` at the end of the line after trimming whitespaces (use `[ ]` to match a literal whitespace at the end).
+* `\` followed by any ASCII alphanumeric character (`[0-9a-zA-Z]`). In particular, C-like escape sequences like `\0`, `\r`, `\n` and `\t` currently are meaningless.
+
+### Quoted identifier
+
+Besides `\`, special characters can also be suppressed by quoting using `"` or `` ` ``.
+
+```
+"db.with.dots"."tbl\1"
+`db.with.dots`.`tbl\2`
+```
+
+The quote characters can be included within an identifier by doubling the character.
+
+```
+"foo""bar".`foo``bar`
+# equivalent to:
+foo\"bar.foo\`bar
+```
+
+Quoted identifier cannot span multiple lines.
+
+It is invalid to partially quote an identifier:
+
+```
+"this is "invalid*.*
+```
+
+### Regular expression
+
+In case very complex rules are needed, each pattern can be written as a regular expression delimited with `/`:
+
+```
+/^db\d{2,}$/./^tbl\d{2,}$/
+```
+
+These regular expressions use the [Go dialect](https://pkg.go.dev/regexp/syntax?tab=doc). The pattern is matched if the identifier contains a substring matching the regular expression. For instance, `/b/` matches `db01`.
+
+> **Note:**
+>
+> Every `/` in the regex must be escaped as `\/`, including inside `[…]`. You cannot place an unescaped `/` between `\Q…\E`.
+
+## Multiple rules
+
+When a table name matches none of the rules in the filter list, the default behavior is to ignore such unmatched tables.
+
+To build a block list, an explicit `*.*` must be used as the first rule, otherwise all tables will be excluded.
```

Comment:
The meaning of this sentence is that, using `!*.Password` without a `*.*` before it is likely an error because all tables are excluded. Using `;` sounds strange 🤔

Reply:
Got it. Thank you for explaining 😃